lib/ansible/modules/cloud/docker/docker_image.py

```diff
@@ -859,11 +869,15 @@ def detect_build_target(client):
     def detect_use_config_proxy(client):
         return client.module.params['build'] and client.module.params['build'].get('use_config_proxy') is not None
 
+    def detect_use_etc_hosts(client):
```

Comment:
```suggestion
    def detect_etc_hosts(client):
```
(Your option is not called `use_etc_hosts`, but `etc_hosts`. You also have to update the `option_minimal_versions` entry below.)